src/structures/mode.jl

```diff
@@ -147,9 +222,64 @@ Base.@kwdef struct PipeLinepackSimple <: PipeMode
     trans_loss::TimeProfile
     opex_var::TimeProfile
     opex_fixed::TimeProfile
-    energy_share::Float64 # Storage energy capacity relative to pipeline capacity
-    directions::Int = 1     # 1: Unidirectional only for pipeline
-    data::Vector{Data} = Data[]
+    energy_share::Float64
+    directions::Int
+    data::Vector{<:Data}
+
+    function PipeLinepackSimple(
+        id::String,
+        inlet::EMB.Resource,
+        outlet::EMB.Resource,
+        consuming::EMB.Resource,
+        consumption_rate::TimeProfile,
+        trans_cap::TimeProfile,
+        trans_loss::TimeProfile,
+        opex_var::TimeProfile,
+        opex_fixed::TimeProfile,
+        energy_share::Float64,
+        data::Vector{<:Data}
```

Comment:
Should there be a comma at the end of this line to have the same syntax as for the next constructor? Maybe run JuliaFormatter => format(".")?
```suggestion
        data::Vector{<:Data},
```